pkg/plugin/framer.go

```diff
@@ -160,3 +161,40 @@ func ExtractDataSets(namePrefix string, rawData map[string]interface{}) ([]Sentr
 	}
 	return sets, nil
 }
+
+func ConvertMetricsResponseToFrame(frameName string, metrics sentry.MetricsResponse) (*data.Frame, error) {
+	if len(metrics.Intervals) == 0 {
+		return data.NewFrameOfFieldTypes(frameName, 0), nil
+	}
+	frame := data.NewFrameOfFieldTypes(frameName, len(metrics.Intervals))
+	field := data.NewField("Timestamp", nil, metrics.Intervals)
+	frame.Fields = append(frame.Fields, field)
+	for _, group := range metrics.Groups {
```

Comment:
This is quite a large block of complex code, does it make sense to maybe break some of this code out into separate functions?

Reply:
I based this on the existing `ConvertStatsV2ResponseToFrame` function in this file which is ~50% larger. Don't see that many pieces that would greatly benefit extracting, maybe field name extraction.